### ISSUE TITLE
Implement a buddy allocator

### ISSUE THREAD

**Issue by tunamayo04** [zerox029/toast#2]
The current slab allocator is very bare-bones and breaks easily. It should be replaced with a better buddy allocator. This process has been started already but is dependent on #1 .

**Comment by tunamayo04** [comment]
Duplicate of #1, heap allocation will be done with a different allocator
